### DISCUSSION TITLE
Allow expression vars to override raw parameters

### DISCUSSION THREAD

**Discussion by andsip** [bigskysoftware/htmx#2320] [Q&A]
This change was introduced in f2941b7a4bf.

I am wondering what was the reasoning behind it, the commit message unfortunately does not tell.

If we take the following example:

```html
<div hx-vals='{"key1": "val1", "key2": "val2"}'>
  <input type="text" name="key1" hx-get="/some/path" hx-trigger="keyup delay:200ms">
</div>
```

If we enter value `"input-val"` into the input text field here, the following request is executed: `/some/path?key1=val1&key2=val2`.

What I would expect and need here though is that the value of the input field is sent, i.e. `/some/path?key1=input-val&key2=val2`.

I went through the source of the htmx library, and I think this is not possible with the current state of the library.
Putting a `hx-vals="unset"` on the `input` field would also not achieve what I need, as I still need `key2` to be sent alongside.

---

Does anyone know why this change was made back then?
Does anyone do know a way how to achieve, what I'm describing?

**Comment by andsip** [comment]
A possible way to solve this could be the following:
```html
<div hx-vals='{"key1": "val1", "key2": "val2"}'>
  <input type="text" name="key1" hx-get="/some/path" hx-trigger="keyup delay:200ms" hx-vals='js:{key1: event.target.value}'>
</div>
```

But for this to work with the `delay:200ms`, #1658 would need to be fixed.

**Comment by Telroshan** [comment]
> the commit message unfortunately does not tell.
> Does anyone know why this change was made back then?

The commit you referenced links to PR #232
Looking at this, it looks like it might have been an unexpected side effect. Seems the PR was about overriding sibling element's values when clicking on a submit button. I admit it looks more logical to me that in your example, the child element's value would override the parent's `hx-vars`
